sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventHubBufferedProducerClient.cs

```diff
@@ -2204,21 +2202,40 @@ private async Task RunProducerManagementAsync(CancellationToken cancellationToke
                                 // Responsibility for releasing the guard semaphore is passed to the task.
 
                                 activeTasks.Add(PublishBatchToPartition(partitionState, releaseGuard: true, activeOperationCancellationSource.Token));
-                                consecutiveEmptyBufferCount = 0;
                             }
 
-                            // If there are no publishing tasks active, introduce a small delay to avoid a tight loop.  If no events are
-                            // available in the buffers, use an increasing delay to avoid wasting resources during periods of low activity.
+                            // If there are no events in the buffer, avoid a tight loop by blocking to wait for events to be enqueued
+                            // after a small delay.
```

Comment:
Just to confirm - small delay referring to the extra tick of the loop?

Reply:
It's an extra tick after we perform a 100ms delay; this is intended to (hopefully) offset the race between going idle and enqueuing an event by allowing a yield point.